api/src/test/java/com/graphhopper/util/shapes/Circle_DiffblueTest.java

```diff
@@ -0,0 +1,653 @@
+/**
+ *
+ *
+ * Copyright 2018 Diffblue Limited
+ *
+ * Diffblue Limited licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.graphhopper.util.shapes;
+
+import com.diffblue.deeptestutils.Reflector;
+import com.graphhopper.util.DistanceCalc3D;
+import com.graphhopper.util.shapes.BBox;
+import com.graphhopper.util.shapes.Circle;
+import com.graphhopper.util.shapes.Shape;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class Circle_DiffblueTest {
```

Comment:
👎 - Discard all tests in this class.  It uses setField too much, I would imagine this would actually be good in the newest release

Reply:
👎  unfortunately

Reply:
Can you create a bug ticket for this as a challenge to instantiate this non-reflectively?